report_presentation/Presenter.py

```diff
@@ -7,4 +16,274 @@ def __init__(self, report_root, initial_warnings, final_warnings, changes):
+            if i.warning_code not in unique_warning_code:
+                unique_warning_code.append(i.warning_code)
+            if i.file_path not in unique_file_path:
+                unique_file_path.append(i.file_path)
+        return unique_warning_code, unique_file_path
+
+    def get_string_of_list(self, alist):
+        astring = "    "
+        for i in alist:
+            astring += i
+            astring += " ;   "
+        return astring
+
+    def create_html(self, dpct_root, destination_root, all_warnings, final_warnings, changes, unique_warning_code, unique_file_path):
+        diff_path = Path.joinpath(Path.cwd(), 'html_files')
+        if diff_path.is_dir() and os.listdir(diff_path):
+            diff_html.remove_diff_folder(diff_path)
+        diff_html.find_dpcpp(dpct_root, destination_root, diff_path)
+
+        with open('report.html', 'w') as report:
+            report.write(
+                Presenter.html_page(self, all_warnings, final_warnings, changes, unique_warning_code, unique_file_path,
+                                    diff_path))
+        with open('subgraphs.html', 'w') as sub_images:
+            sub_images.write(sub_graph.add_images())
+
+    def remove_image_folder(self, image_path):
+        for i in os.listdir(image_path):
+            os.remove(os.path.join(image_path, i))
+        os.rmdir(image_path)
+
+    def show_visualize(self, report_root, all_warnings):
+        image_path = Path.joinpath(report_root, 'images')
+        if image_path.is_dir() and os.listdir(image_path):
+            Presenter.remove_image_folder(self, image_path)
+        image_path.mkdir(parents=True, exist_ok=True)
+        graph.visualization_overall(all_warnings, image_path)
+        graph.visualization_partial(all_warnings, image_path)
+
+
+def get_warnings_and_changes(dpct_root, destination_root):
+    preAnalyser = PreAnalyser(dpct_root)
+    all_warnings = preAnalyser.get_all_warnings()
+    postAnalyser = PostAnalyser(destination_root)
+    final_warnings = postAnalyser.get_all_warnings()
+    changes = postAnalyser.get_all_recommendation()
+    return all_warnings, final_warnings, changes
+
+
+def run_presenter():
+    report_root = Path(__file__).parent
+    cta_path = Path(__file__).parent.parent.resolve()
+    dpct_root = Path.joinpath(cta_path, 'auto_editor', 'sample_data', 'test_project')
+    destination_root = Path.joinpath(cta_path, 'auto_editor', 'sample_data', 'destination_dir')
+    all_warnings, final_warnings, changes = get_warnings_and_changes(dpct_root, destination_root)
+
+    presenter = Presenter(report_root, all_warnings, final_warnings, changes)
+    unique_warning_code, unique_file_path = presenter.get_unique_filepath_and_warning_code(all_warnings)
+
+    presenter.show_visualize(report_root, all_warnings)
+    presenter.create_html(dpct_root, destination_root, all_warnings, final_warnings, changes, unique_warning_code, unique_file_path)
+
+
+if __name__ == '__main__':
+    run_presenter()
```

Comment:
Maybe a separate script is better, since this has no more use once the tool is used as expected